Support options to skip resize and optimize

ono opened the pull request:
Support options to skip resize and optimize. You may want to skip those steps on CI environment to speed up tests.

ono committed:
Support options to skip resize and optimize. You may want to skip
those steps on CI environment to spped up tests.

callumacrae commented on tasks/responsive-images.js:
```diff
@@ -16,7 +17,7 @@ module.exports = function (gulp, plugins, options) {
       .pipe(handleRename('-xlarge'))
       .pipe(handleChanged())
-      .pipe(handleOptimize())
+      .pipe(gulpif(!options.skipOptimize, handleOptimize()))
```
This is already available in `plugins.if`, no need to require it :)

callumacrae commented:
Can you document this?

callumacrae commented:
Also looks like the test failure is an actual failure, in iojs. Not sure if it's related to what you've done, I can take a look if you want.

ono committed:
Add notes about skipResize and skipOptimize options

ono commented:
@callumacrae sure. fixed with `plugins.if` and added the document for the options.

yeah, i'm aware of the test failures but it doesn't seem to be related to my change. it'd be great if you could look into it. thanks!

callumacrae commented:
The build passed that time :weary: 

I've updated the version of iojs on master. If it continues randomly failing, I'll add iojs to `allow_failures`.

ono commented:
@callumacrae Yeah, it could be conditional. I found even gulp-svg2png doesn't compare the png with binary. https://github.com/akoenig/gulp-svg2png/blob/master/specs/plugin.spec.ts

callumacrae commented:
this was merged, not sure why github didn't notice
